openwhisk/actionProxy.go

```diff
@@ -71,7 +72,14 @@ func NewActionProxy(baseDir string, compiler string, outFile *os.File, errFile *
 
 //SetEnv sets the environment
 func (ap *ActionProxy) SetEnv(env map[string]interface{}) {
-	ap.env["__OW_API_HOST"] = os.Getenv("__OW_API_HOST")
+	// propagate all the variables starting with "__OW_"
+	for _, v := range os.Environ() {
+		if strings.HasPrefix(v, "__OW_") {
```

Comment:
i guess why not re-export all env variables?

Reply:
Because for example in Kubernetes environment variables are used for service discovery and this way an action could read them and get access to some informations that it is better they do not see. I do not know how the KubernetesFactory created action exectors but I guessed it was better to be safe and only propagate enviroment variables somewhat "marked" to be exposed. Is it a reason good enough?